config/opal_compare_semver.m4

```diff
@@ -0,0 +1,59 @@
+dnl -*- shell-script -*-
+dnl
+dnl Copyright (c) 2020-2022 Cisco Systems, Inc.  All rights reserved.
+dnl
+dnl $COPYRIGHT$
+dnl
+dnl Additional copyrights may follow
+dnl
+dnl $HEADER$
+dnl
+
+dnl This macro compares two "semver" version numbers.
+dnl See https://semver.org/
+dnl
+dnl Arguments:
+dnl
+dnl $1: semver A
+dnl $2: semver B
+dnl $3: shell variable name to store the result
+dnl
+dnl The value stored in the result shell variable will be:
+dnl
+dnl "same": if A == B
+dnl "lesser": if A < B, in a semver comparison
+dnl "greater": if A > B, in a semver comparison
+dnl
+AC_DEFUN([OPAL_COMPARE_SEMVER],[
```

Comment:
Because mine is BETTER!

😝 

I actually looked for a macro like this before writing OPAL_COMPARE_SEMVER, but your Google-fu and/or memory is better than mine.  😄